types/block_test.go

```diff
@@ -458,84 +466,82 @@ func TestMaxHeaderBytes(t *testing.T) {
 }
 
 func randCommit(now time.Time) *Commit {
-	lastID := makeBlockIDRandom()
 	h := int64(3)
 	voteSet, _, vals := randVoteSet(h-1, 1, tmproto.PrecommitType, 10, 1)
-	commit, err := MakeCommit(lastID, h-1, 1, voteSet, vals, now)
+	bid := randBlockID()
+	commit, err := MakeCommit(bid, h-1, 1, voteSet, vals, now)
 	if err != nil {
 		panic(err)
 	}
 	return commit
 }
 
-func hexBytesFromString(s string) bytes.HexBytes {
-	b, err := hex.DecodeString(s)
-	if err != nil {
-		panic(err)
-	}
-	return bytes.HexBytes(b)
-}
-
-func TestBlockMaxDataBytes(t *testing.T) {
-	testCases := []struct {
-		maxBytes      int64
-		valsCount     int
-		evidenceBytes int64
-		panics        bool
-		result        int64
-	}{
-		0: {-10, 1, 0, true, 0},
-		1: {10, 1, 0, true, 0},
-		2: {851, 1, 0, true, 0},
-		3: {852, 1, 0, false, 0},
-		4: {853, 1, 0, false, 1},
-		5: {964, 2, 0, false, 1},
-		6: {1063, 2, 100, false, 0},
-	}
-
-	for i, tc := range testCases {
-		tc := tc
-		if tc.panics {
-			assert.Panics(t, func() {
-				MaxDataBytes(tc.maxBytes, tc.evidenceBytes, tc.valsCount)
-			}, "#%v", i)
-		} else {
-			assert.Equal(t,
-				tc.result,
-				MaxDataBytes(tc.maxBytes, tc.evidenceBytes, tc.valsCount),
-				"#%v", i)
-		}
-	}
+func TestRandCommit(t *testing.T) {
+	// err := randBlockID().ValidateBasic()
+	err := randCommit(time.Now()).ValidateBasic()
+	require.NoError(t, err)
 }
 
-func TestBlockMaxDataBytesNoEvidence(t *testing.T) {
-	testCases := []struct {
-		maxBytes  int64
-		valsCount int
-		panics    bool
-		result    int64
-	}{
-		0: {-10, 1, true, 0},
-		1: {10, 1, true, 0},
-		2: {851, 1, true, 0},
-		3: {852, 1, false, 0},
-		4: {853, 1, false, 1},
-	}
-
-	for i, tc := range testCases {
-		tc := tc
-		if tc.panics {
-			assert.Panics(t, func() {
-				MaxDataBytesNoEvidence(tc.maxBytes, tc.valsCount)
-			}, "#%v", i)
-		} else {
-			assert.Equal(t,
-				tc.result,
-				MaxDataBytesNoEvidence(tc.maxBytes, tc.valsCount),
-				"#%v", i)
-		}
-	}
-}
+// func TestBlockMaxDataBytes(t *testing.T) {
```

Comment:
What's the status of the tests below?

Reply:
this is another test that I commented out rather than fix, as checks for max bytes should be reevaluated. 

I definitely need to keep track of this in an issue if we decide to keep these commented out.

Curious as to what @liamsi thinks.

Reply:
@adlerjohn, I feel like I asked this before, but anyway it seems like we need to superficially discuss the following again. Are there any constraints for max block size?

Reply:
The constraint on maximum block size is whatever the erasure coding library supports. I have to look into exactly what it is, but at the very least a 128*128 original square with 256 bytes per share should work.

Reply:
While I think everything testing for Maximum bytes of X will be easier after #287, as we will not have to account for the max data availability header, this particular case is strongly related to #77

Reply:
> Curious as to what @liamsi thinks.

I think it should not be too difficult to figure out the numbers but it is OK to do this in the followup pr as you said (once instead twice). And yes, best to track it in an issue to make sure we won't forget 👍🏼